game/utils.js

```diff
@@ -75,3 +75,7 @@ function setsEqual(a, b) {
   assert(b instanceof Set);
   return a.size === b.size && [...a].every((item) => b.has(item));
 }
+
+function delay(time) {
+  return new Promise(resolve => setTimeout(resolve, time));
+}
```

Comment:
Added delay to the progress saving process. Otherwise, it undoes your last move.